### PR TITLE
Add a step-by-step flow to illustrate nonce validation.

### LINKED ISSUE #403: Add step by step example for private transaction nonce validation
See comments on https://github.com/hyperledger/besu-docs/pull/387

The two scenarios are: 

a. Private transaction with correct nonce: 
    1. Submit private transaction with nonce 3 (it's the 4th transaction in the privacy group from this account) 
   2. Private transaction distributed to all participants. 
   3. Privacy marker transaction created and submitted to transaction pool with nonce of 0 if using one time accounts. 
       If using a specific account (specified using --privacy-marker-transaction-signing-key-file), the nonce for that account is obtained and used for the privacy marker transaction. 
  4. Privacy marker transaction mined and included in block.
  5. When block containing privacy marker transaction is imported and privacy marker transaction processed, the private transaction is retrieved from Orion and executed. The nonce is validated as correct at this point. 

b. Private transaction with incorrect nonce: 
   1. Submit private transaction with nonce 2 (it's the 4th transaction in the privacy group from this account) 
   2. Private transaction distributed to all participants. 
   3. Privacy marker transaction created and submitted to transaction pool with nonce of 0 if using one time accounts. 
       If using a specific account (specified using --privacy-marker-transaction-signing-key-file), the nonce for that account is obtained and used for the privacy marker transaction. 
  4. Privacy marker transaction mined and included in block.
  5. When block containing privacy marker transaction is imported and privacy marker transaction processed, the private transaction is retrieved from Orion and executed. The private transaction execution will fail because the private transaction nonce is incorrect.

### PR DESCRIPTION
**Pull request by bgravenorst**
## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [ ] you fixed all the issues raised by the tests, if any.
- [ ] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

Add a step-by-step flow to illustrate nonce validation.

## Issue fixed

Fixes #403 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### CODE REVIEW

File: docs/Concepts/Privacy/Private-Transactions.md [hunk 1]
```diff
@@ -79,6 +81,23 @@ for the private transaction with the incorrect nonce.
     of nonce management when sending multiple private transactions. The example calculates the
     correct nonces for the private transactions and privacy marker transactions outside of Besu.
 
+The following private transaction flow illustrates when nonce validation occurs:
+
+1. Submit a private transaction with a [nonce value](#private-transaction-nonce).
+1. The private transaction is distributed to all participants in the privacy group.
+1. The privacy marker transaction is created and submitted to the transaction pool with a nonce of `0`
+    if using one time accounts. If using a specific account using
```
Comment (alexandratran) [inline docs/Concepts/Privacy/Private-Transactions.md:89]:
```suggestion
    if using one time accounts. If using a specific account with
```

File: docs/Concepts/Privacy/Private-Transactions.md [hunk 2]
```diff
@@ -79,6 +81,23 @@ for the private transaction with the incorrect nonce.
...
+    [`--privacy-marker-transaction-signing-key-file`](../../Reference/CLI/CLI-Syntax.md#privacy-marker-transaction-signing-key-file), the
+    nonce for that account is obtained and used for the privacy marker transaction.
+1. The privacy marker transaction is mined and included in the block.
+1. When the block containing the privacy marker transaction is imported, and privacy marker
```
Comment (alexandratran) [inline docs/Concepts/Privacy/Private-Transactions.md:93]:
```suggestion
1. After the block containing the privacy marker transaction is imported and the privacy marker
```

**Review by alexandratran — APPROVED** [review]
Minor suggestions.

**Review by lucassaldanha — APPROVED** [review]
LGTM.

However, most of the confusion about privacy nonces is around how sending multiple private txs in sequence requires the user to manage the nonce themselves. Usually, they will create the PMT themselves instead of letting Besu do it. 

It might be worth considering adding a section that explain the process of sending private txs using (`priv_distributeRawTransaction` and `eth_sendRawTransaction`).

**Pull request merged**